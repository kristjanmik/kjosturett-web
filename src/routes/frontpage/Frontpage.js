import React, { PureComponent } from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import parties from '../../../data/build/parties.json';
import Link from '../../Link';

import s from './Frontpage.scss';
import Party from '../../components/Party';
import PartyGrid from '../../components/PartyGrid';

class FrontPage extends PureComponent {
  render() {
    const showElectionDayMessage =
      Date.now() > 1632528000000 && Date.now() < 1632614400000;

    return (
      <div>
        {showElectionDayMessage && (
          <div className={s.voteCTA}>
            <p>Kæri kjósandi, til hamingju með daginn!</p>
            <div className={s.ctaButtons}>
              <Link href="/kosningaprof" className={s.button}>
                1. Taktu kosningaprófið
              </Link>
              <div className={s.arrow}>→</div>
              <Link href="/malefni/atvinnumal" className={s.button}>
                2. Kynntu þér málefnin
              </Link>
              <div className={s.arrow}>→</div>
              <Link href="/kjorskra" className={s.button}>
                3. Finndu út hvar þú átt að kjósa
              </Link>
            </div>
          </div>
        )}
        <PartyGrid>
          {parties.map(party => (
            <Party
              {...party}
              key={party.letter}
              href={`/flokkur/${party.url}`}
            />
          ))}
        </PartyGrid>
      </div>
    );
  }
}

export default withStyles(s)(FrontPage);
