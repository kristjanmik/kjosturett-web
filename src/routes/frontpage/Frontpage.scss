@import '../../components/variables.scss';

.voteCTA {
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid #efefef;
  margin-bottom: 1em;
  padding: 1em;
  font-size: 1.1rem;
  text-align: center;
  background: #fafafa;

  .ctaButtons {
    display: flex;
    margin-top: 10px;

    $breakPoint: 850px;

    @media (max-width: $breakPoint) {
      flex-direction: column;
    }

    .button {
      padding: 0.45em 0.8em;
      font-weight: 500;
      cursor: pointer;
      text-decoration: none;
      color: #343434;

      @media (max-width: $breakPoint) {
        padding: 0.3em 0.5em;
      }
    }

    .arrow {
      display: flex;
      align-items: center;
      margin: 0 10px;

      @media (max-width: $breakPoint) {
        display: none;
      }
    }

    .button:hover {
      background: #eaeaea;
    }
  }
}
