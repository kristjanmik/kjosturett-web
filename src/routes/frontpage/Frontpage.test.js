import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';
import Frontpage from './Frontpage';
import { mount } from 'enzyme';

describe('Frontpage', () => {
  test('renders frontpage correctly', () => {
    const wrapper = renderer
      .create(
        <App
          context={{
            insertCss: () => {},
            fetch: () => {},
            store: {
              subscribe: () => {},
              dispatch: () => {},
              getState: () => {},
            },
          }}
        >
          <Frontpage />
        </App>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
  test('has 10 parties', () => {
    const wrapper = mount(
      <App
        context={{
          insertCss: () => {},
          fetch: () => {},
          store: {
            subscribe: () => {},
            dispatch: () => {},
            getState: () => {},
          },
        }}
      >
        <Frontpage />
      </App>
    );

    const partyCount = wrapper.find('.party').length;

    expect(partyCount).toBe(11);
  });
});
