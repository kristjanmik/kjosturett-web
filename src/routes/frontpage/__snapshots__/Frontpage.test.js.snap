// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontpage renders frontpage correctly 1`] = `
<div>
  <div
    className="grid"
  >
    <a
      className="party"
      href="/flokkur/framsoknarflokkurinn"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Framsókn's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/framsoknarflokkurinn.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Framsókn
        </h3>
        <p
          className="leader"
        >
          Sigurður Ingi Jóhannsson
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          B
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/vidreisn"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Viðreisn's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/vidreisn.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Viðreisn
        </h3>
        <p
          className="leader"
        >
          Þorgerður Katrín Gunnarsdóttir
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          C
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/sjalfstaedisflokkurinn"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Sjálfstæðisflokkurinn's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/sjalfstaedisflokkurinn.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Sjálfstæðisflokkurinn
        </h3>
        <p
          className="leader"
        >
          Bjarni Benediktsson
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          D
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/flokkur-folksins"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Flokkur Fólksins's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/flokkur-folksins.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Flokkur Fólksins
        </h3>
        <p
          className="leader"
        >
          Inga Sæland
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          F
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/sosialistaflokkurinn"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Sósíalistaflokkurinn's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/sosialistaflokkurinn.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Sósíalistaflokkurinn
        </h3>
        <p
          className="leader"
        >
          Gunnar Smári Egilsson
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          J
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/lydraedisflokkurinn"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Lýðræðisflokkurinn's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/lydraedisflokkurinn.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Lýðræðisflokkurinn
        </h3>
        <p
          className="leader"
        >
          Arnar Þór Jónsson
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          L
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/midflokkurinn"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Miðflokkurinn's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/midflokkurinn.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Miðflokkurinn
        </h3>
        <p
          className="leader"
        >
          Sigmundur Davíð Gunnlaugsson
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          M
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/piratar"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Píratar's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/piratar.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Píratar
        </h3>
        <p
          className="leader"
        >
          Þórhildur Sunna Ævarsdóttir
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          P
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/samfylkingin"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Samfylkingin's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/samfylkingin.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Samfylkingin
        </h3>
        <p
          className="leader"
        >
          Kristrún Frostadóttir
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          S
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/vinstri-graen"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Vinstri Græn's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/vinstri-graen.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Vinstri Græn
        </h3>
        <p
          className="leader"
        >
          Svandís Svavarsdóttir
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          V
        </span>
      </span>
    </a>
    <a
      className="party"
      href="/flokkur/abyrg-framtid"
      onClick={[Function]}
    >
      <span
        className="imgWrap"
      >
        <img
          alt="Ábyrg Framtíð's logo"
          className="image"
          src="https://assets.kjosturett.is/party-icons/abyrg-framtid.png"
        />
      </span>
      <span
        className="info"
      >
        <h3
          className="name"
        >
          Ábyrg Framtíð
        </h3>
        <p
          className="leader"
        >
          Jóhannes Loftsson
        </p>
        <span
          className="letter"
        >
          <span>
            x
          </span>
          Y
        </span>
      </span>
    </a>
  </div>
</div>
`;
